code/game/objects/structures/tables_racks.dm

```diff
@@ -297,10 +297,12 @@
 /obj/structure/table/glass/proc/table_shatter(mob/M)
 	visible_message("<span class='warning'>[src] breaks!</span>",
 		"<span class='danger'>You hear breaking glass.</span>")
-	playsound(src.loc, "shatter", 50, 1)
-	for(var/i in debris)
-		var/atom/movable/AM = i
-		AM.loc = get_turf(src)
+	var/turf/T = get_turf(src)
+	playsound(T, "shatter", 50, 1)
+	for(var/I in debris)
+		var/atom/movable/AM = I
+		AM.forceMove(T)
+		debris -= AM

```

Comment:
To be more specific, Destroy() deletes the debris we just moved to the turf, because it's still in the debris list.